docs/source/reference/api-server/api-server-upgrade.rst

```diff
@@ -0,0 +1,204 @@
+
+.. note::
+
+    Upgrading the API server introduces downtime. We recommend to schedule the upgrade during a maintenance window. If you have a staging environment, you can upgrade the staging environment first to make sure the upgrade is successful before upgrading the production environment.
+
+Prerequisites
+-------------
+
+* Complete :ref:`sky-api-server-deploy` to deploy the API server.
+
+Step 1: Prepare an upgrade
+--------------------------
+
+1. Get the image you want to upgrade to from SkyPilot dockerhub repository: `nightly release <https://hub.docker.com/r/berkeleyskypilot/skypilot-nightly/tags>`_. A specific tag other than ``latest`` is required to make the upgrade process predictable.
+2. Update SkyPilot helm repository to the latest version:
+
+.. code-block:: bash
+
+    helm repo update skypilot
+
+3. Prepare variables for the upgrade, the ``NAMESPACE`` and ``RELEASE_NAME`` should be the same as the previous deployment step:
+
+.. code-block:: bash
+
+    NAMESPACE=<installed-namespace>
+    RELEASE_NAME=<installed-release-name>
+    # e.g. IMAGE=berkeleyskypilot/skypilot-nightly:1.0.0-dev.20250406
+    IMAGE=<image-from-dockerhub>
+
+Step 2: Cordon the API server
+------------------------------------------
+
+To minimize the impact of the downtime introduced by the upgrade, we recommend condon the API server before upgrading the API server. The following steps require `patch` and `exec` (or `port-forward`) access to the API server Pod.
+
+1. Patch the API server to remove it from the Ingress backend:
+
+.. note::
+
+    All new requests will be rejected by the Ingress after this step. Make sure there is no critical service depending on the API server before proceeding.
+
+.. code-block:: bash
+
+    kubectl get pod -l app=${RELEASE_NAME}-api -oname | xargs kubectl patch --type merge -p '{"metadata": {"labels": {"skypilot.co/ready": null}}}'
+
+2. Verify the API server is cordoned, you should see the following error:
+
+.. code-block:: console
+
+    $ sky api info
+    sky.exceptions.ApiServerConnectionError: Could not connect to SkyPilot API server at <URL>. Please ensure that the server is running. Try: curl <URL>
+
+.. dropdown:: Resolve cordon failure for early 0.8.0 nightly release
+
+    If you are upgrading from early 0.8.0 nightly release that does not support cordoning (``sky api info`` will succeed), you can manually enable cordon support by running:
+
+    .. code-block:: bash
+
+        kubectl patch service ${RELEASE_NAME}-api-service -p '{"spec":{"selector":{"skypilot.co/ready":"true"}}}'
+    
+    After the patch, verify the API server is cordoned again.
+
+3. For requests that have been sent to the API server, check the status of the requests and wait for the critical ones to finish:
```

Comment:
```suggestion
3. Drain the old API server by waiting or canceling current requests:

```

How about we make `Check requests` and `Cancel requests` in two different tabs to save some vertical space?